pandas/io/excel/_util.py

```diff
@@ -24,13 +25,15 @@ def register_writer(klass):
     for ext in klass.supported_extensions:
         if ext.startswith('.'):
             ext = ext[1:]
-        if ext not in _writer_extensions:
+        if ext not in _registered_writer_extensions:
```

Comment:
So do we even need this block or is this entirely a no-op? I don't think we have any reader support for things outside of the already registered extensions right?

Reply:
As far as I can tell it is a no-op, but I can't judge the implications of removing it. The tests pass without this.

Reply:
If it simplifies code and doesn't impact functionality (which I believe to be the case) then I'd say just delete

Reply:
I stand corrected, `test_register_writer` fails for some builds after the code removal, so I went ahead and removed the test too. Now I am a bit less sure if removing this is a good idea, I'll look into the history to see why this was added in the first place

Reply:
What was the failure? On a quick look I think it's only failing because of the `test` extension but I can't imagine that being an actual use case anyway. Would probably just remove that one particular extension

Reply:
So the test was added in https://github.com/pandas-dev/pandas/commit/42e06a78e6639ed5a73035d3890b9c051bf08d1d, with no specific usecase in mind, just to "Make ExcelWriter more pluggable"

Reply:
Right - I think your removal here is too aggressive. Should only need to remove the non-excel extension not the entire test

Reply:
I believe the test is about being able to register a custom excel writer with custom extensions. The PR as is now breaks the ability to register custom extensions, so I removed the test for that.  I read a bit on excel extensions, apparently there is also the binary`.xlsb` extensions that excel supports natively. And contrary to what I expected, there is even some limited (read only) python support with the pyxlsb library

Reply:
Keep in mind these mechanisms are private. We don't currently support xlsb in any fashion (you can search issue tracker for requests) so the reasoning for removal of the registration here is arguably that we've over-engineered this feature internally. The test case for a `.test` extension is rather contrived

Reply:
Ok, thanks for the backgrounds. The test is now failing for Windows py27_np121. I don't understand the error, I'm guessing it's just flakiness. Any way to rerun the a specific job?